### PR TITLE
doc(quick) : improvements in order to improve the experience of new users

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
- Add prerequisites
- Small changes in order to make easier the understatement for new users. 

**Motivation**
IMHO: These small changes are very helpful to make it easier for new users and avoid mistake and/or them face issues because they were missing the prerequisites for example. Which motivate me to do this PR was my experience as a new user trying to follow up these steps.  
* https://github.com/kubernetes-sigs/kubebuilder/issues/984
* https://github.com/kubernetes-sigs/kubebuilder/issues/1034
* https://github.com/kubernetes-sigs/kubebuilder/issues/1033
* https://github.com/kubernetes-sigs/kubebuilder/issues/1031

### CODE REVIEW

File: docs/book/src/quick-start.md [hunk 1]
```diff
@@ -7,9 +7,19 @@ This Quick Start guide will cover:
 - [Running locally](#test-it-out-locally)
 - [Running in-cluster](#run-it-on-the-cluster)
 
+## Prerequisites
+
+- [go](https://golang.org/dl/) version v1.12+.
+- [docker](https://docs.docker.com/install/) version 17.03+.
+  - Alternatively [podman](https://github.com/containers/libpod/blob/master/install.md) `v1.2.0+` or [buildah](https://github.com/containers/buildah/blob/master/install.md) `v1.7+`
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]: we don't provide podman and buildah instructions, so I'm hesitant to include these specifically.  Perhaps `(or some other way to build and run containers)` or something similar.

File: docs/book/src/quick-start.md [hunk 2]
```diff
@@ -7,9 +7,19 @@ This Quick Start guide will cover:
 - [Running locally](#test-it-out-locally)
 - [Running in-cluster](#run-it-on-the-cluster)
 
+## Prerequisites
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]: maybe just merge this with installation?  `kubebuilder` isn't exactly a prerequisite, since we show how to install it below.

File: docs/book/src/quick-start.md [hunk 3]
```diff
@@ -24,60 +34,53 @@ sudo mv /tmp/kubebuilder_2.0.0_${os}_${arch} /usr/local/kubebuilder
...
-in that API group-version:
+* Run the following command to create a new API (group/version) as `webapp/v1` and the new Kind(CRD) `Guestbook` on it:
 
 ```bash
 kubebuilder create api --group webapp --version v1 --kind Guestbook
 ```
 
-This will create the files `api/v1/guestbook_types.go` and
-`controller/guestbook_controller.go` for you to edit.
+**NOTE:** This will create the files `api/v1/guestbook_types.go` where the API is defined
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]: make this a note block along the same lines as the others (look for `<aside class="note">`)

File: docs/book/src/quick-start.md [hunk 4]
```diff
@@ -24,60 +34,53 @@ sudo mv /tmp/kubebuilder_2.0.0_${os}_${arch} /usr/local/kubebuilder
...
-This will create the files `api/v1/guestbook_types.go` and
-`controller/guestbook_controller.go` for you to edit.
+**NOTE:** This will create the files `api/v1/guestbook_types.go` where the API is defined 
+and the `controller/guestbook_controller.go` where the reconciliation business logic is implemented for this Kind(CRD) if you choose yes for the both options.  
 
-**Optional:** Edit the API definition or the reconciliation business
-logic. For more on this see [What's in
-a Controller](cronjob-tutorial/controller-overview.md) and [Designing an
-API](/cronjob-tutorial/api-design.md).
+* (**OPTIONAL**) Edit the API definition and the reconciliation business
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]: why is this in parens now?

File: docs/book/src/quick-start.md [hunk 5]
```diff
@@ -24,60 +34,53 @@ sudo mv /tmp/kubebuilder_2.0.0_${os}_${arch} /usr/local/kubebuilder
 export PATH=$PATH:/usr/local/kubebuilder/bin
 ```
 
-You can also install a KubeBuilder master snapshot from
-`https://go.kubebuilder.io/dl/latest/${os}/${arch}`.
+**NOTE:** Also, you can install a master snapshot from `https://go.kubebuilder.io/dl/latest/${os}/${arch}`.
 
-Install [kustomize](https://sigs.k8s.io/kustomize/docs/INSTALL.md) v3.1.0+
+* Install [kustomize](https://sigs.k8s.io/kustomize/docs/INSTALL.md) v3.1.0+
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]: why did you change all the bullet points below?

File: docs/book/src/quick-start.md (hunk 3 above)
Comment (DirectXMan12) [reply docs/book/src/quick-start.md]: ditto elsewere in your changes

File: docs/book/src/quick-start.md (hunk 5 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: To make easier see/identify that has more than one step to follow up and this part does not be skipped by mistake.

File: docs/book/src/quick-start.md (hunk 2 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: Make sense remove kubebuilder

File: docs/book/src/quick-start.md (hunk 1 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: Done

File: docs/book/src/quick-start.md (hunk 2 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: The kubebuilder makes sense not be in the prerequisite,  however, the others one shows that should be.

File: docs/book/src/quick-start.md (hunk 4 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: Removed 👍

File: docs/book/src/quick-start.md (hunk 5 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: After changes, it makes no longer sense.  Tks

Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: Solved

**Comment by camilamacedo86** [comment]
Hi @DirectXMan12,

I think with the changes made now are addressing the main goal of  #1034 too. Besides it not be done exactly in this way.

File: docs/book/src/quick-start.md (hunk 3 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: I tried to use : 

```
<aside class="notes" data-markdown>
{note}
</aside>
```
and it did not work. I mean, was the same as nothing. I never ever saw it before. How it supose render? Should it add the NOTE?

File: docs/book/src/quick-start.md [hunk 6]
```diff
@@ -7,9 +7,17 @@ This Quick Start guide will cover:
...
+- [go](https://golang.org/dl/) version v1.12+.
+- [docker](https://docs.docker.com/install/) version 17.03+.
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) version v1.11.3+.
+- [kustomize](https://sigs.k8s.io/kustomize/docs/INSTALL.md) v3.1.0+
+- Access to a Kubernetes v1.11.3+ cluster.
+
 ## Installation
 
-Install [kubebuilder](https://sigs.k8s.io/kubebuilder):
+Following the steps to install [kubebuilder](https://sigs.k8s.io/kubebuilder):
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]: This is only part of a sentence now, and doesn't really make sense in isolation.  `Install` or `Follow` is an imperative -- it gives someone a command.  This form states what you're doing, and generally in this structure you'd expect it to be followed by another clause (e.g. `Following the steps to install KubeBuilder, Agnes ran go get`, `Agnes installed kubebuilder by following the steps below`).  Alternative, the phrase `The following` is short for `The following <plural noun>`, but you can't leave off the `The`.

Please switch back to an imperative form.

File: docs/book/src/quick-start.md [hunk 7]
```diff
@@ -24,64 +32,67 @@ sudo mv /tmp/kubebuilder_2.0.1_${os}_${arch} /usr/local/kubebuilder
...
-in that API group-version:
+* Run the following command to create a new API (group/version) as `webapp/v1` and the new Kind(CRD) `Guestbook` on it:
 
 ```bash
 kubebuilder create api --group webapp --version v1 --kind Guestbook
 ```
 
-This will create the files `api/v1/guestbook_types.go` and
-`controller/guestbook_controller.go` for you to edit.
+**Note:** If you press `y` for Create Resource [y/n] and for Create Controller [y/n] then this will create the files `api/v1/guestbook_types.go` where the API is defined
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]: same before about a proper note block.  Look for `class="note` in the documentation to find an example.

File: docs/book/src/quick-start.md [hunk 8]
```diff
@@ -24,64 +32,67 @@ sudo mv /tmp/kubebuilder_2.0.1_${os}_${arch} /usr/local/kubebuilder
...
-You can also install a KubeBuilder master snapshot from
-`https://go.kubebuilder.io/dl/latest/${os}/${arch}`.
-
-Install [kustomize](https://sigs.k8s.io/kustomize/docs/INSTALL.md) v3.1.0+
+**Note:** Also, you can install a master snapshot from `https://go.kubebuilder.io/dl/latest/${os}/${arch}`.
 
 ## Create a Project
 
-Initialize a new project and Go module for your controllers:
+Create a directory then run the following command inside of it to initialize a new project:
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]:
```suggestion
Create a directory, and then run the following command inside of it to initialize a new project:
```

File: docs/book/src/quick-start.md [hunk 9]
```diff
@@ -24,64 +32,67 @@ sudo mv /tmp/kubebuilder_2.0.1_${os}_${arch} /usr/local/kubebuilder
...
+**Note:** If you press `y` for Create Resource [y/n] and for Create Controller [y/n] then this will create the files `api/v1/guestbook_types.go` where the API is defined 
+and the `controller/guestbook_controller.go` where the reconciliation business logic is implemented for this Kind(CRD) if you choose yes for the both options.  
 
-**Optional:** Edit the API definition or the reconciliation business
-logic. For more on this see [What's in
-a Controller](cronjob-tutorial/controller-overview.md) and [Designing an
-API](/cronjob-tutorial/api-design.md).
+* **OPTIONAL:** Edit the API definition and the reconciliation business
+logic. For more info see [Designing an API](/cronjob-tutorial/api-design.md) and [What's in
+a Controller](cronjob-tutorial/controller-overview.md). Following an example.
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]: Ditto from above -- you can't leave off the `The` here:

```suggestion
a Controller](cronjob-tutorial/controller-overview.md).  For example:
```

File: docs/book/src/quick-start.md [hunk 10]
```diff
@@ -24,64 +32,67 @@ sudo mv /tmp/kubebuilder_2.0.1_${os}_${arch} /usr/local/kubebuilder
...
 
-**Optional:** Edit the API definition or the reconciliation business
-logic. For more on this see [What's in
-a Controller](cronjob-tutorial/controller-overview.md) and [Designing an
-API](/cronjob-tutorial/api-design.md).
+* **OPTIONAL:** Edit the API definition and the reconciliation business
+logic. For more info see [Designing an API](/cronjob-tutorial/api-design.md) and [What's in
+a Controller](cronjob-tutorial/controller-overview.md). Following an example. 
+
+```go (api/v1/guestbook_types.go)
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]: put this in a details block.

File: docs/book/src/quick-start.md [hunk 11]
```diff
@@ -24,64 +32,67 @@ sudo mv /tmp/kubebuilder_2.0.1_${os}_${arch} /usr/local/kubebuilder
...
 
-## Test It Out Locally
+## Test It Out 
 
 You'll need a Kubernetes cluster to run against.  You can use
 [KIND](https://sigs.k8s.io/kind) to get a local cluster for testing, or
 run against a remote cluster.
 
-Your controller will automatically use the current context in your
-kubeconfig file (i.e. whatever cluster `kubectl cluster-info` shows).
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]: don't drop this block

Comment (DirectXMan12) [reply docs/book/src/quick-start.md]: but you can reword it to say that `make install` will use whatever context

File: docs/book/src/quick-start.md [hunk 12]
```diff
@@ -24,64 +32,67 @@ sudo mv /tmp/kubebuilder_2.0.1_${os}_${arch} /usr/local/kubebuilder
...
-
-Install the CRDs into the cluster:
-
+* Install the CRDs into the cluster:
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]: note that if you're adding bullet points, you should make sure the fenced code block is indented to line up with the start of the text so that the code block joins the list item properly.

Comment (DirectXMan12) [reply docs/book/src/quick-start.md]: Ditto everywhere else in this doc

File: docs/book/src/quick-start.md [hunk 13]
```diff
@@ -90,19 +101,19 @@ kubectl apply -f config/samples/
...
 
 ```bash
-make deploy IMG=<some-registry>/controller
+make deploy IMG=<some-registry>/<project-name>:tag
 ```
 
-If you encounter RBAC errors, you may need to grant yourself cluster-admin
-privileges:
+**Note:** If you encounter RBAC errors, you may need to grant yourself cluster-admin
+privileges or be logged in as admin. For further information see an example over how to [create a sample user](https://github.com/kubernetes/dashboard/blob/master/docs/user/access-control/creating-sample-user.md#creating-sample-user).
```
Comment (DirectXMan12) [inline docs/book/src/quick-start.md]: is there another link we can use?  Dashboard isn't the best place to link to considering its state.   Since this usually happens on GKE, maybe link here: https://cloud.google.com/kubernetes-engine/docs/how-to/role-based-access-control#iam-rolebinding-bootstrap

File: docs/book/src/quick-start.md (hunk 6 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: Tks English is not my first language and this explanation was very helpful,

File: docs/book/src/quick-start.md (hunk 9 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: 👍

File: docs/book/src/quick-start.md (hunk 11 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: 👍

File: docs/book/src/quick-start.md (hunk 7 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: 👍

File: docs/book/src/quick-start.md (hunk 12 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: 👍

File: docs/book/src/quick-start.md (hunk 13 above)
Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: Done 👍

File: docs/book/src/quick-start.md [hunk 14]
```diff
@@ -24,64 +32,139 @@ sudo mv /tmp/kubebuilder_2.0.1_${os}_${arch} /usr/local/kubebuilder
...
 ```bash
 kubebuilder create api --group webapp --version v1 --kind Guestbook
 ```
 
-This will create the files `api/v1/guestbook_types.go` and
-`controller/guestbook_controller.go` for you to edit.
+<aside class="note">
+<h1>Press Options</h1>
+
+If you press `y` for Create Resource [y/n] and for Create Controller [y/n] then this will create the files `api/v1/guestbook_types.go` where the API is defined
```
Comment (erictune) [inline docs/book/src/quick-start.md]: I understood the quickstart to be giving the user a tour of the system.  Therefore, perhaps it should direct the user to press `y` twice button rather than giving them a choice.

File: docs/book/src/quick-start.md [hunk 15]
```diff
@@ -24,64 +32,139 @@ sudo mv /tmp/kubebuilder_2.0.1_${os}_${arch} /usr/local/kubebuilder
...
-
 ```bash
 make run
 ```
 
-## Install Samples
+## Install Instances of Custom Resources
 
-Create your samples (make sure to edit them first if you've changed the
+If you pressed `y` for Create Resource [y/n] then you created an (CR)Custom Resource for your (CRD)Custom Resource Definition in your samples (make sure to edit them first if you've changed the
```
Comment (erictune) [inline docs/book/src/quick-start.md]: Since this is giving the user a tour of the system, suggest avoiding "If" statements.

Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: But it is optional. IMHO: it is not mandatory for the user to perform the quick start. However, at the end of it was added a suggestion for the user to follow up the tutorial where all these points will be detailed explained. Are you ok with?

Comment (erictune) [reply docs/book/src/quick-start.md]: I would think that 95% or more of use cases for KubeBuilder involve both a CRD and a controller.  
Chosing not to use them is sort of an advanced use case.  So a quickstart (which I see as like a demo), does not need to cover an advanced use case.

Fewer choices for a user in introductory instructions is better.

Comment (camilamacedo86) [reply docs/book/src/quick-start.md]: IMHO the goal is to show how to create, build and deploy a project to give a macro idea and not actually cover a basic use case. The basic demo/use case is covered by the tutorial and not here.
